Example of of Crypto Deterministic Config using https://cloud.google.…

hilliao opened the pull request:
…com/dlp/docs/pseudonymization#supported-methods to resolve https://github.com/googleapis/python-dlp/issues/108

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dlp/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<108> 🦕

tmatsuo commented on samples/snippets/deid.py:
```diff
@@ -288,6 +288,93 @@ def deidentify_with_fpe(
+        key_name=None,
+        wrapped_key=None,
+):
+    """Uses the Data Loss Prevention API to deidentify sensitive data in a
```
How about just:

```python
    """Deidentifies sensitive data in a string using deterministic encryption."""
```
?

tmatsuo commented on samples/snippets/deid.py:
```diff
@@ -288,6 +288,93 @@ def deidentify_with_fpe(
+
+    # The wrapped key is base64-encoded, but the library expects a binary
+    # string, so decode it here.
+    import base64
```
How about to move this import statement just before the dlp import?

```python
    import base64

    # Import the client library
    import google.cloud.dlp
```

tmatsuo commented on samples/snippets/deid.py:
```diff
@@ -380,6 +467,92 @@ def reidentify_with_fpe(
+        key_name=None,
+        wrapped_key=None,
+):
+    """Uses the Data Loss Prevention API to reidentify sensitive data in a
```
Maybe similar change for the docstring?

tmatsuo commented on samples/snippets/deid.py:
```diff
@@ -380,6 +467,92 @@ def reidentify_with_fpe(
+
+    # The wrapped key is base64-encoded, but the library expects a binary
+    # string, so decode it here.
+    import base64
```
Ditto

tmatsuo reviewed (approved):
LGTM, with few small suggestions

tmatsuo commented:
In case you don't have access to the build log, here is the lint output:

```
./deid.py:471:1: E303 too many blank lines (3)
# [START dlp_reidentify_deterministic]
^
./deid.py:472:1: E302 expected 2 blank lines, found 3
def reidentify_with_deterministic(
^
```

hilliao commented:
I fixed the 5 places and fixed parameters in the unit tests. I had to push an amended commit as new commits failed https://www.conventionalcommits.org/en/v1.0.0/ checks. Should new commit's comment be identical to prior commit's comment? What's the rule @tmatsuo ?

tmatsuo commented:
@hilliao
The conventional commit check seems good now.

tmatsuo commented:
@hilliao Do you have access to the build log?
https://source.cloud.google.com/results/invocations/8de21aa8-ef52-4544-8e1b-93d75a7bac21/log

hilliao commented:
I can see the build error but I don't have access to test_deidentify_with_deterministic 's output to see the output of the tokenized harmful string. I believe  the error from the build is caused by using my own KMS key instead of projects/python-docs-samples-tests/locations/global/keyRings/dlp-test/cryptoKeys/dlp-test which I don't have access to.
build error:
`
debug_error_string = "{"created":"@1617263155.197209081","description":"Error received from peer ipv4:74.125.197.95:443","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Could not transform all content due to transformation errors. To force this request to succeed anyway (by skipping any transformations that cause errors), set `transformation_error_handling.mode` to `leave_untransformed` in the DeidentifyConfig; this will return the partially transformed content along with an error summary.\n
`
Lack of access:
```shell
echo -ne $RAW_KEY | gcloud kms encrypt --key $KEY_NAME --keyring $KEY_RING_NAME --location global --plaintext-file=- --ciphertext-file wrapped.key --project python-docs-samples-tests && cat wrapped.key | base64
ERROR: (gcloud.kms.encrypt) PERMISSION_DENIED: Permission 'cloudkms.cryptoKeyVersions.useToEncrypt' denied on resource 'projects/python-docs-samples-tests/locations/global/keyRings/dlp-test/cryptoKeys/dlp-test' (or it may not exist).

```
I believe the tokenized string is in the output of test_deidentify_with_deterministic but I can't find its output from [the build error](https://source.cloud.google.com/results/invocations/8de21aa8-ef52-4544-8e1b-93d75a7bac21/targets). Can you tell me the tokenized string or the whole build output so I can update  [the code line](https://github.com/googleapis/python-dlp/pull/119/files#diff-ff9a25f9e4ca9982289a3961329a05137fe5fc750a4dea5d0338edb1c793d33dR226)?
```python
labeled_fpe_string = "My SSN is SSN_TOKEN(36):AUGFopzf7RiKAHWD2Sm5GJWS0Yfezsbc1OU="
```

tmatsuo commented:
@hilliao I got it. I'll try to add a commit.

tmatsuo commented:
Re: conventional commits
I also got an error. My guess is our bot can not parse your commit message well. I squashed the commit.

tmatsuo commented:
Now the tests are passing. It might be better to get the string dynamically with a purest fixture.

tmatsuo commented:
@hilliao WDYT?

hilliao commented:
I would like to merge this pull request first; then I will create a new issue to change the `re-identification` test method to `de-identification then re-identification` in the test method. Can you merge?

tmatsuo commented:
@hilliao Sounds good

tmatsuo commented:
@hilliao Thanks for your contribution!
